### PR TITLE
Replace merge statements for optional parameters

### PR DESCRIPTION
**Pull request by marvinfriede**
Applying the suggestions from #201.

### CODE REVIEW

File: src/dftd4/param.f90 [hunk 1]
```diff
@@ -67,7 +67,8 @@ subroutine get_rational_damping(functional, param, s9)
    integer :: is, id
    logical :: mbd
 
-   mbd = merge(s9 /= 0.0_wp, .true., present(s9))
+   mbd = .false.
+   if (present(s9) .and. s9 /= 0.0_wp) mbd = .true.
```
Comment (e-kwsm) [inline src/dftd4/param.f90:71]: The condition may not be short-circuited.

```suggestion
   if (present(s9)) mbd = s9 /= 0.0_wp
```

Comment (marvinfriede) [reply src/dftd4/param.f90:71]: You are right. Thanks for the hint.

File: src/dftd4/param.f90 [hunk 2]
```diff
@@ -67,7 +67,8 @@ subroutine get_rational_damping(functional, param, s9)
    integer :: is, id
    logical :: mbd
 
-   mbd = merge(s9 /= 0.0_wp, .true., present(s9))
+   mbd = .false.
```
Comment (awvwgk) [inline src/dftd4/param.f90:70]: This changes the default?

File: src/dftd4/param.f90 [hunk 3]
```diff
@@ -117,11 +118,20 @@ pure function dftd_param(s6, s8, a1, a2, alp) result(param)
       real(wp), intent(in) :: s8, a1, a2
       real(wp), intent(in), optional :: s6, alp
       type(rational_damping_param) :: param
+      real(wp) :: s6_local, alp_local, s9_local
+
+      s6_local = 1.0_wp
+      if (present(s6)) s6_local = s6
+      s9_local = 1.0_wp
```
Comment (awvwgk) [inline src/dftd4/param.f90:125]: Changing the default here as well?

File: src/dftd4/param.f90 [hunk 4]
```diff
@@ -117,11 +118,20 @@ pure function dftd_param(s6, s8, a1, a2, alp) result(param)
       real(wp), intent(in) :: s8, a1, a2
       real(wp), intent(in), optional :: s6, alp
       type(rational_damping_param) :: param
+      real(wp) :: s6_local, alp_local, s9_local
```
Comment (awvwgk) [inline src/dftd4/param.f90:121]: I think s6_ would be sufficient.

**Review by awvwgk — CHANGES_REQUESTED** [review]

File: src/dftd4/param.f90 (hunk 3 above)
Comment (marvinfriede) [reply src/dftd4/param.f90]: Good catch, I changed it back to `0.0`.

File: src/dftd4/param.f90 (hunk 4 above)
Comment (marvinfriede) [reply src/dftd4/param.f90]: Done.

File: src/dftd4/param.f90 (hunk 2 above)
Comment (marvinfriede) [reply src/dftd4/param.f90:70]: Why does that matter? The following check for `s9` will always be evaluated (to true if s9 is set).

Comment (awvwgk) [reply src/dftd4/param.f90:70]: This changes which method is used in case s9 is not provided.

Comment (marvinfriede) [reply src/dftd4/param.f90:70]: What about this? Should be the translation of the merge-statement.
```fortran
if (s9 /= 0.0_wp) then
  mbd = .true.
else
  mbd = present(s9)
endif
```

Comment (awvwgk) [reply src/dftd4/param.f90:70]: That does still change the behavior. See [merge](https://fortran-lang.org/en/learn/intrinsics/array/#merge).

Why not just use
```f90
mbd = .true.
if (present(s9)) mbd = s9 /= 0.0_wp
```

Comment (marvinfriede) [reply src/dftd4/param.f90:70]: That works. (I thought `merge` takes the mask as first argument like np.where)

**Review by awvwgk — APPROVED** [review]

**Pull request merged**